### DISCUSSION TITLE
How to migrate existing Rules/Actions/Worklfows from StackStorm running in a Azure VM to Kubernetes HA Cluster?

### DISCUSSION THREAD

**Discussion by Bindu-yadav8** [StackStorm/st2#5618] [Operations]
Hello,

We are using StackStorm version st2 3.4.1, deployed on a VM and are managing and running Actions & workflows on the VM. 

Now we have deployed Highly Available StackStorm cluster based on Kubernetes and wanted to migrate all the existing actions/workflows and rules into the K8 cluster.

Can anybody provide a series of steps and suggestions on how to proceed on this migration?

@armab Could you please provide some insights on this?

**Comment by arm4b** [comment]
Is there something that you already researched and tried?

What prevents you from creating a new staging environment where you'd migrate all the old actions and workflows and start testing in there?
